src/utils.js

```diff
@@ -19,40 +19,38 @@ So we either need to use a new instance of a regex everywhere.
 */
 var sourceMapUrlRegEx = function(){ return /\/\/\# sourceMappingURL\=.*/g;};
 
+var commentFormatters = {
+  css: function cssCommentFormatter(preLine, newline, url) {
+    return preLine + "/*# sourceMappingURL=" + url + " */" + newline;
+  },
+  js: function jsCommentFormatter(preLine, newline, url) {
+    return preLine + "//# sourceMappingURL=" + url + newline;
+  },
+  'default': function defaultFormatter() {
+    return '';
+  }
+}
+
 
 var getCommentFormatter = function (file) {
   var extension = file.relative.split('.').pop(),
-    fileContents =  file.contents.toString(),
-    newline =  detectNewline.graceful(fileContents || ''),
-    commentFormatter = function(url) {
-      return '';
-    };
+    fileContents = file.contents.toString(),
+    newline = detectNewline.graceful(fileContents || '');
 
-  if (file.sourceMap.preExistingComment){
-    debug(function() { return 'preExistingComment commentFormatter'; });
-    commentFormatter = function(url) {
-      return "//# sourceMappingURL=" + url + newline;
-    };
-    return commentFormatter;
-  }
+  var commentFormatter = commentFormatters.default;
 
-  switch (extension) {
-    case 'css':
-      debug(function() { return 'css commentFormatter';});
-      commentFormatter = function(url) {
-        return newline + "/*# sourceMappingURL=" + url + " */" + newline;
-      };
-      break;
-    case 'js':
-      debug(function() { return 'js commentFormatter'; });
-      commentFormatter = function(url) {
-        return newline + "//# sourceMappingURL=" + url + newline;
-      };
-      break;
-    default:
-      debug(function() { return 'unknown commentFormatter'; });
+  if (file.sourceMap.preExistingComment) {
+    commentFormatter = (commentFormatters[extension] || commentFormatter).bind(undefined, '', newline)
+    debug(function () {
+      return 'preExistingComment commentFormatter ' + commentFormatter.name;
+    });
+  } else {
+    commentFormatter = (commentFormatters[extension] || commentFormatter).bind(undefined, newline, newline);
   }
 
+  debug(function () {
```

Comment:
It seems like this should be moved into the `else {}` block just above?  Unless you want the `!!file.sourceMap.preExistingComment` to print two debug messages.

Reply:
@coreyfarrell I see you are practaclly the lead on `istanbul-api` and `nyc` can you upgrade us to the latest as we're on such an old version of istanbul? Would the new versions still work in node 0.8 and 0.12?

Reply:
The current versions of istanbul and nyc all require node.js 6, otherwise I would have done this already.

To be honest I'm hoping that `gulp@5` will drop node.js versions that are EOL.  The need to avoid dependencies which use ES6 is increasingly difficult.

Reply:
100% agree thanks